File: 백준/Bronze/17293. 맥주 99병/맥주 99병.py
import sys

input = sys.stdin.readline
print = sys.stdout.write

N = int(input().strip())

def S(num):
    if num == 1:
        return '1 bottle'
    elif num > 1:
        return f'{num} bottles'
    elif num == 0:
        return 'no more bottles'

for i in range(N, 0, -1):
    print(f"{S(i)} of beer on the wall, {S(i)} of beer.\nTake one down and pass it around, {S(i-1)} of beer on the wall.\n\n")

print(f"No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, {S(N)} of beer on the wall.\n")

File: 백준/Bronze/17293. 맥주 99병/맥주 99병.java
import java.io.*;

public class Main {
    private static String S(int num){
        if(num == 1) return "1 bottle";
        else if(num == 0) return "no more bottles";
        else if(num > 1) return num + " bottles";
        else return "error";
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine().trim());
        br.close();

        for(int i = N; i > 0; i--){
            sb.append(S(i)).append(" of beer on the wall, ")
            .append(S(i)).append(" of beer.\nTake one down and pass it around, ")
            .append(S(i-1)).append(" of beer on the wall.\n\n");
        }

        sb.append("No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, ")
        .append(S(N)).append(" of beer on the wall.");

        System.out.println(sb.toString());
    }
}